tests/cli/bootprep/test_vars.py

```diff
@@ -0,0 +1,127 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Unit tests for variable loading
+"""
+
+from textwrap import dedent
+import unittest
+from unittest.mock import mock_open, patch
+
+from sat.cli.bootprep.constants import LATEST_VERSION_VALUE
+from sat.cli.bootprep.vars import VariableContext
+
+
+class TestEnumeratingVars(unittest.TestCase):
+    """Test enumerating variables and sources"""
+
+    def setUp(self):
+        self.context = VariableContext()
+        self.vars_file_contents = dedent("""
+        foo:
+            version: 2.3.4
+        bar:
+            version: 4.5.6
+        """)
+
+        self.mock_catalog = patch('sat.cli.bootprep.vars.HPCSoftwareRecipeCatalog').start()
+        self.software_recipe_vars = {
+            'foo': {'version': '1.2.3'},
+            'bar': {'version': '2.3.4'}
+        }
+        self.mock_catalog.return_value.get_latest_version.return_value.all_vars = self.software_recipe_vars
+
+    def tearDown(self):
+        patch.stopall()
+
+    def test_getting_latest_recipe_vars(self):
+        """Test enumerating variables from the latest recipe"""
+        self.context.recipe_version = LATEST_VERSION_VALUE
+        expected_elements = {('foo.version', '1.2.3', 'latest recipe'),
+                             ('bar.version', '2.3.4', 'latest recipe')}
+        for element in list(self.context.enumerate_vars_and_sources()):
+            self.assertIn(element, expected_elements)
+
+    def test_getting_arbitrary_recipe_vars(self):
+        """Test enumerating variables from a given recipe version"""
+        self.context.recipe_version = '22.07'
+        expected_elements = {('foo.version', '1.2.3', '22.07 recipe'),
+                             ('bar.version', '2.3.4', '22.07 recipe')}
+        for element in list(self.context.enumerate_vars_and_sources()):
+            self.assertIn(element, expected_elements)
+
+    def test_getting_file_vars(self):
+        """Test retrieving variables from a vars file"""
+        self.software_recipe_vars = {}
+        vars_file_path = 'some_path.yaml'
+        self.context.vars_file_path = vars_file_path
+        with patch('sat.cli.bootprep.vars.open', mock_open(read_data=self.vars_file_contents)):
+            expected_elements = {('foo.version', '2.3.4', vars_file_path),
+                                 ('bar.version', '4.5.6', vars_file_path)}
+            for element in list(self.context.enumerate_vars_and_sources()):
+                self.assertIn(element, expected_elements)
+
+    def test_getting_cli_vars(self):
+        """Test retrieving variables from the command line"""
+        self.software_recipe_vars = {}
```

Comment:
I don't think this is actually working as expected. It is still not getting any variables from the recipe, but the`HPCSoftwareRecipeCatalog.get_latest_version` method is not being called. The `get_recipe_version` method is being called because `self.context.recipe_version` is not set to `LATEST_VERSION_VALUE`.

Furthermore, this doesn't change the value of `self.mock_catalog.return_value.get_latest_version.return_value.all_vars` because it's already been set in `setUp`.